File: ants/c++/ants.cpp
#include <iostream>
#include <algorithm>

using namespace std;
void slove();
// ===== ↓↓↓input↓↓↓ ===== //
const int MAX_N = 3;
int L = 10;
int n = MAX_N;
int x[MAX_N] = {2, 6, 7};
// ===== ↑↑↑input↑↑↑ ===== //

int main(int argc, char *argv[]) {
	slove();
}
void slove() {
	// min time
	int min_time = 0;
	for (int i = 0; i < n; i++) {
		min_time = std::max(min_time, std::min(x[i], L - x[i]));
	}
	// max time
	int max_time = 0;
	for (int i = 0; i < n; i++) {
		max_time = std::max(max_time, std::max(x[i], L - x[i]));
	}
	
	printf("%d %d\n", min_time, max_time);
}

File: warm_up/ants/go/ants.go
package main

import (
	"fmt"
	"math"
)

func main() {
	// ===== ↓↓↓input↓↓↓ ===== //
	var L float64
	L = 10
	var n = 3
	x := []float64{2, 6, 7}
	// ===== ↑↑↑input↑↑↑ ===== //

	// min time
	var min float64
	min = 0
	for i := 0; i < n; i++ {
		min = math.Max(min, math.Min(x[i], L-x[i]))
	}

	// max time
	var max float64
	max = 0
	for i := 0; i < n; i++ {
		max = math.Max(max, math.Max(x[i], L-x[i]))
	}

	fmt.Println(min)
	fmt.Println(max)
}
